test/libsolidity/semanticTests/externalContracts/ramanujan_pi.sol

```diff
@@ -33,8 +33,8 @@ contract test {
 }
 // ----
 // constructor()
-// gas irOptimized: 79503
-// gas irOptimized code: 328000
+// gas irOptimized: 77762
+// gas irOptimized code: 307000
```

Comment:
The `TFviu` fix for `create_memory_array.sol` (https://github.com/ethereum/solidity/pull/14928#discussion_r1543737156) destroys this gain and makes the cost increase by 8% here instead.

I see now that it's actually all coming from constructor - neither the new sequence nor the one with the fix seems to have affected execution cost of the `prb_pi()` function. So it's all just bytecode size.

Interestingly, the new sequence (without the fix) produces smaller bytecode even thought the Yul source is visibly larger. There's a whole chunk of ~125 lines that's not present in the output from default sequence. It does have one function less, but the extra chunk does not comes from inlining of that function. The function is a specialization of another, present in both versions. Overall the bytecode size difference is ~100 bytes.

One thing that draws attention when comparing assembly is that the bigger version produced by the default sequence has more large constants in it. My theory is that this is the major factor affecting the bytecode size for this contract. It aligns with my observation that versions with bigger bytecode have more copies of the specialized function I mentioned above. The function has several large numbers in it:

```yul
            function fun_mulDivFixedPoint(var_x, var_y) -> var_result
            {
                let usr$mm := mulmod(var_x, var_y, not(0))
                let var_prod0 := mul(var_x, var_y)
                let _1 := lt(usr$mm, var_prod0)
                let _2 := sub(usr$mm, var_prod0)
                let var_prod1 := sub(_2, _1)
                let _3 := 1000000000000000000
                let var_remainder := mulmod(var_x, var_y, _3)
                let var_roundUpUnit := gt(var_remainder, 499999999999999999)
                if eq(_2, _1)
                {
                    var_result := add(div(var_prod0, _3), var_roundUpUnit)
                    leave
                }
                if iszero(gt(_3, var_prod1)) { revert(0, 0) }
                var_result := add(mul(or(shr(18, sub(var_prod0, var_remainder)), shl(238, sub(var_prod1, gt(var_remainder, var_prod0)))), 78156646155174841979727994598816262306175212592076161876661508869554232690281), var_roundUpUnit)
           }
```

The code optimized with default sequence has two copies (the original and a specialization).
The one from the new sequence (-6% gas) has one. 
The one from the new sequence with the `TFviu` fix (+8% gas) has three.

My conclusion here is that this is again a problem with the specializer. Unfortunately this time I don't have a good fix. I can tweak the sequence to make the specialization more or less aggressive but some contracts gain while others lose. I think that dealing with this properly would require some heuristics that can decide when it's worth it to specialize. Or maybe some adjustment to inlining heuristics - those specialized functions do not get inlined and perhaps they should be.

In the light of that I'm not sure if I should really treat the original 9% increase for `create_memory_array.sol` as a regression. It's balanced out by the 6% decrease for `ramanujan_pi.sol` and I think it might also balance out for real contracts. That's pretty much what happens in external tests.